### ISSUE TITLE
Space Type Measure Tags stability and functionality 

### ISSUE THREAD

**Issue by DavidGoldwasser** [NREL/OpenStudio#3120]
![screen shot 2018-05-04 at 2 26 10 pm](https://user-images.githubusercontent.com/4857055/39652817-250b75d8-4fac-11e8-96e5-9e74bef4447c.png)

There are a number of bugs with this sub-tab, for simplicity, I'm going to file them here, with separate check boxes. 

- [ ] If you edit the "Standards Building Type" or "Standards Space Type" field, and then leave the space type tabs, without fist activating the name field, or another sub-tab within space type OpenStudio will crash. I tested this in both 2.5.1 and 2.4.0 on mac.
- [ ] When you change the "Standards Building Type" the "Standard Space Type" choices don't update until you leave the sub-tab and come back. Tested on 2.5.1. and 2.4.0 on mac
- [ ] When you type a string in the "Standards Building Type" or "Standards Space Type" fields it only lets you enter a single character, which you have to leave the sub-tab and come back to see (you can actually now enter a second character, then leave and come back to enter third). Tested on 2.5.1. and 2.4.0 in mac.
- [ ] 2.5.1 seems to have some but not all of the new "Standards Building Types" and "Standards Space Types" from the latest openstudio-standards gem. On the attached screenshot for example you can see what I assume are DEER building types, but you also see "Deli/Bakery" as a single space type, but in the gem there is now a "Deli" space type and a "Bakery" space type. Similar to this the office space types in the gem are now "SmallOffice - OpenOffice" instead of "OpenOffice" but the app in 2.5.1 doesn't reflect this. I ran across this in updating the Space Type and Construction Set Wizard, but this is also relevant to our model articulation measures used elsewhere.

**Comment by jmarrec** [comment]
**Tested on Ubuntu 16.04** with current develop (10b655b3a) built in **Debug**

1. If you edit the "Standards Building Type" or "Standards Space Type" field, and then leave the space type tabs, without fist activating the name field, or another sub-tab within space type OpenStudio will crash. I tested this in both 2.5.1 and 2.4.0 on mac.

**=> Cannot reproduce**

2. When you change the "Standards Building Type" the "Standard Space Type" choices don't update until you leave the sub-tab and come back. Tested on 2.5.1. and 2.4.0 on mac

**=> Can reproduce**

3. When you type a string in the "Standards Building Type" or "Standards Space Type" fields it only lets you enter a single character, which you have to leave the sub-tab and come back to see (you can actually now enter a second character, then leave and come back to enter third). Tested on 2.5.1. and 2.4.0 in mac.

**=> Typing something in the "Standards Building Type" actually produces a hard crash due to hitting an ASSERT statement**

```
[BOOST_ASSERT] <2> Assertion m_choicesMap.find(result) != m_choicesMap.end() failed on line 656 of std::__cxx11::string openstudio::OptionalChoiceConceptImpl<ChoiceType>::get() [with ChoiceType = std::__cxx11::basic_string<char>; std::__cxx11::string = std::__cxx11::basic_string<char>] in file /home/julien/Software/Others/OpenStudio/openstudiocore/src/openstudio_lib/../shared_gui_components/OSConcepts.hpp.
[BOOST_ASSERT] <2> Assertion m_choicesMap.find(result) != m_choicesMap.end() failed on line 656 of std::__cxx11::string openstudio::OptionalChoiceConceptImpl<ChoiceType>::get() [with ChoiceType = std::__cxx11::basic_string<char>; std::__cxx11::string = std::__cxx11::basic_string<char>] in file /home/julien/Software/Others/OpenStudio/openstudiocore/src/openstudio_lib/../shared_gui_components/OSConcepts.hpp.

```


4. `2.5.1` seems to have some but not all of the new "Standards Building Types" and "Standards Space Types" from the latest openstudio-standards gem. On the attached screenshot for example you can see what I assume are DEER building types, but you also see "Deli/Bakery" as a single space type, but in the gem there is now a "Deli" space type and a "Bakery" space type. Similar to this the office space types in the gem are now "SmallOffice - OpenOffice" instead of "OpenOffice" but the app in 2.5.1 doesn't reflect this. I ran across this in updating the Space Type and Construction Set Wizard, but this is also relevant to our model articulation measures used elsewhere.

**==> I still see "Deli/Bakery" under supermarket.**

**Comment by jmarrec** [comment]
3 is likely related to  #2814 (fixed in #3201)

**Comment by jmarrec** [comment]
@asparke2 I sent you a Slack message, I need your help especially on item 4 please, thanks!

**Comment by jmarrec** [comment]
I think this can be closed now via  #3227
